src/App.tsx

```diff
@@ -1,24 +1,265 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useState, useCallback, useEffect, useMemo,
+} from 'react';
+import classNames from 'classnames';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { Todo } from './types/Todo';
+import {
+  getTodos, postTodo, deleteTodo, patchTodo, USER_ID,
+} from './api/todos';
+import { TodoList } from './components/TodoList/TodoList';
+import { FilterType } from './types/FilterType';
+import { Footer } from './components/Footer/Footer';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [disabledInput, setDisabledInput] = useState(false);
+  const [query, setQuery] = useState('');
+  const [errorText, setErrorText] = useState('');
+  const [filterType, setFilterType] = useState(FilterType.All);
+  const [loading, setLoading] = useState(false);
+  const [loadingTodoId, setLoadingTodoId] = useState<number[]>([0]);
+
+  const closeError = () => setTimeout(() => setErrorText(''), 3000);
+  const handleCloseButton = () => setErrorText('');
+  const completedTodos = todos.filter(todo => todo.completed);
+  const activeTodos = todos.filter(todo => !todo.completed);
+
+  const fetchTodos = useCallback(async () => {
+    try {
+      const todosFromServer = await getTodos(USER_ID);
+
+      setTodos(todosFromServer);
+    } catch (error) {
+      setErrorText('Unable to load todos');
+      closeError();
+    }
+  }, []);
+
+  useEffect(() => {
+    fetchTodos();
+  }, [fetchTodos]);
+
+  const addTodo = useCallback(async (title: string) => {
+    try {
+      setLoading(true);
+
+      if (!title.length) {
+        setErrorText("Title can't be empty");
+        closeError();
+
+        return;
+      }
+
+      const newTodo = {
+        userId: USER_ID,
+        title,
+        completed: false,
+      };
+
+      setDisabledInput(true);
+      setTempTodo({ ...newTodo, id: 0 });
+
+      const createdTodo = await postTodo(newTodo);
+
+      setTodos(prevTodos => [...prevTodos, createdTodo]);
+    } catch (error) {
+      setErrorText('Unable to add todo');
+      closeError();
+    } finally {
+      setTempTodo(null);
+      setDisabledInput(false);
+      setLoading(false);
+    }
+  }, []);
+
+  const removeTodo = useCallback(async (id: number) => {
+    try {
+      setLoadingTodoId(state => [...state, id]);
+      await deleteTodo(id);
+      setTodos(todos.filter(todo => todo.id !== id));
+    } catch {
+      setErrorText('Unable to delete a todo');
+      closeError();
+    } finally {
+      setLoadingTodoId([]);
+    }
+  }, [todos]);
+
+  const updateTodo = useCallback(
+    async (id: number, data: string | boolean) => {
```

Comment:
It's not a good practice to use the same variable for such different types (`string | boolean`). Maybe would be better to pass the object { completed, title }? What do you think?

Reply:
please fix this comment.
![image](https://user-images.githubusercontent.com/98764641/232460256-6900e439-452d-4235-99fe-acb3a783459c.png)
ask in the chat if you have questions

Reply:
still not fixed. write in the `fe_chat` if you have difficulties